langchain4j-couchbase/src/main/java/dev/langchain4j/store/embedding/couchbase/CouchbaseEmbeddingStore.java

```diff
@@ -0,0 +1,331 @@
+        Map<String, Object> embedding = new HashMap<>();
+        embedding.put("enabled", true);
+        List<Map<String, Object>> fields = new ArrayList<>();
+        embedding.put("fields", fields);
+
+        Map<String, Object> field = new HashMap<>();
+        fields.add(field);
+        field.put("dims", dimensions);
+        field.put("index", true);
+        field.put("name", "vector");
+        field.put("similarity", "l2_norm");
+        field.put("type", "vector");
+        field.put("vector_index_optimized_for", "recall");
+
+
+        return embedding;
+    }
+
+    private Map<String, Object> docConfig() {
+        Map<String, Object> docConfig = new HashMap<>();
+        docConfig.put("mode", "scope.collection.type_field");
+        docConfig.put("type_field", "type");
+        return docConfig;
+    }
+
+    @Override
+    public String add(Embedding embedding) {
+        String id = UUID.randomUUID().toString();
+        add(id, embedding);
+        return id;
+    }
+
+    @Override
+    public void add(@NonNull String id, @NonNull Embedding embedding) {
+        addInternal(Collections.singletonList(id), Collections.singletonList(embedding), null);
+    }
+
+    @Override
+    public String add(@NonNull Embedding embedding, @Nullable TextSegment textSegment) {
+        return addAll(Collections.singletonList(embedding), textSegment == null ? null : Collections.singletonList(textSegment)).get(0);
+    }
+
+    @Override
+    public List<String> addAll(@NonNull List<Embedding> embeddings) {
+        return addAll(embeddings, null);
+    }
+
+    @Override
+    public List<String> addAll(@NonNull List<Embedding> embeddings, @Nullable List<TextSegment> embedded) {
+        List<String> ids = embeddings.stream()
+                .map(i -> UUID.randomUUID().toString())
+                .collect(Collectors.toList());
+        addInternal(ids, embeddings, embedded);
+        return ids;
+    }
+
+    private void addInternal(List<String> ids, List<Embedding> embeddings, List<TextSegment> embedded) {
+        if (ids == null || embeddings == null || ids.isEmpty() || embeddings.isEmpty()) {
+            return;
+        }
+
+        int size = ids.size();
+        if (embedded != null && embedded.size() != size) {
+            throw new IllegalArgumentException("embedded and ids have different sizes");
+        }
+
+        for (int i = 0; i < size; i++) {
+            Document document = new Document();
+            document.setId(ids.get(i));
+            Embedding embedding = embeddings.get(i);
+            document.setVector(embedding.vector());
+            if (embedded != null) {
+                TextSegment segment = embedded.get(i);
+                document.setText(segment.text());
+                document.setMetadata(segment.metadata().toMap());
+            }
+            collection.upsert(ids.get(i), document);
+        }
+    }
+
+    @Override
+    public void removeAll(Collection<String> ids) {
```

Comment:
inte test of removing (`extends EmbeddingStoreWithRemovalIT`) shall be added then